### DISCUSSION TITLE
Redirect Loop Between /admin and /login Routes When Integrating Auth0 with FilamentPHP

### DISCUSSION THREAD

**Discussion by abdulmlik** [filamentphp/filament#15061] [General]
### Package

filament/filament

### Package Version

v3.2.0

### Laravel Version

v11.9

### PHP Version

8.2

### Problem description

I'm facing an issue while integrating Auth0 with FilamentPHP in my Laravel application. I've followed the [Auth0 Laravel documentation](https://github.com/auth0/laravel-auth0/blob/main/docs/Users.md) to set up authentication. However, when attempting to access the Filament admin panel (/admin route), I'm caught in a redirect loop between the /admin and /login routes.

### Expected behavior

The user should be authenticated via Auth0 and redirected to the FilamentPHP admin panel.

### Steps to reproduce

Steps to Reproduce:
Set up a Laravel project with FilamentPHP.
Follow the [Auth0 Laravel guide](https://github.com/auth0/laravel-auth0/blob/main/docs/Users.md) to integrate Auth0.
Attempt to access the FilamentPHP admin panel.
auth.php (Laravel Configuration):
```
'guards' => [
        'auth0' => [
            'driver' => 'session',
            'provider' => 'auth0-provider',
        ],
],

'providers' => [
     'auth0-provider' => [
            'driver' => 'auth0.provider',
            // 'model' => env('AUTH_MODEL', App\Models\User::class),
            'repository' => \App\Repositories\UserRepository::class,
        ],
],
```
UserRepository.php:
```
<?php

namespace App\Repositories;

use App\Models\User;
use Illuminate\Contracts\Auth\Authenticatable;
use Auth0\Laravel\{UserRepositoryAbstract, UserRepositoryContract};


final class UserRepository extends UserRepositoryAbstract implements  UserRepositoryContract
{
    public function fromAccessToken(array $user): ?Authenticatable
    {

        return User::where('auth0', $user['sub'])->firstOrFail();
    }

    public function fromSession(array $user): ?Authenticatable
    {

        $userdb = User::where('email', $user['email'])->firstOrFail();
        if($userdb != null && $userdb->auth0 == null) {
            $userdb->auth0 = $user['sub'] ?? '';
            $userdb->save();
        }

        return $userdb;
    }
}
```
AdminPanelProvider.php:
```
<?php

namespace App\Providers\Filament;

class AdminPanelProvider extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->id('admin')
            ->path('admin')
             ...
            // ->login(Login::class)
            ->authMiddleware([
                Authenticate::class,
            ])
            ->authGuard('auth0');
    }
}
```

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/abdulmlik/FilamentPHP-auth0

### Donate 💰 to fund this issue

- You can donate funding to this issue. We receive the money once the issue is completed & confirmed by you.
- 100% of the funding will be distributed between the Filament core team to run all aspects of the project.
- Thank you in advance for helping us make maintenance sustainable!

<a href="https://polar.sh/filamentphp/filament/issues/13895">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/filamentphp/filament/issues/13895/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/filamentphp/filament/issues/13895/pledge.svg">
</picture>
</a>

**Comment by danharrin** [comment]
Since this is not an officially supported integration, I think I will convert this into a discussion.

**Comment by jhoanborges** [comment]
I was doing this until i found crashes between routes. Not sure about a workaround.

**Comment by G3z** [comment]
I did Manage to fix this:
I created a new middleware for filament

```php
<?php

namespace App\Http\Middleware;

use Filament\Facades\Filament;
use Filament\Models\Contracts\FilamentUser;
use Illuminate\Auth\Middleware\Authenticate as Middleware;
use Illuminate\Http\Request;

class AdminAuthMiddleware extends Middleware
{
	/**
	 * Handle an incoming request.
	 */
	protected function authenticate($request, array $guards): void
	{
		if (!auth()->check()) {
			$this->unauthenticated($request, $guards);

			return; /* @phpstan-ignore-line */
		}

		/** @var Model $user */
		$user = auth()->user();

		$panel = Filament::getCurrentPanel();

		abort_if(
			$user instanceof FilamentUser ?
				(!$user->canAccessPanel($panel)) :
				('local' !== config('app.env')),
			403,
		);
	}

	protected function redirectTo($request): ?string
	{
		return Filament::getLoginUrl();
	}
}
```


and specified it in the AdminPanelProvider

```
// ->login() disabled 
->middleware([
...
\Auth0\Laravel\Middleware\AuthenticatorMiddleware::class,  \\ add the middleware from the auth0/login package
AuthenticateSession::class, // I put mine above this one
...
])
->authMiddleware([
    AdminAuthMiddleware::class,
])
```
